### ISSUE TITLE
Key error in aligning at segment level after aligning at word level

### ISSUE THREAD

**Issue by krayush07** [A2Zadeh/CMU-MultimodalSDK#122]
This is the errorstack:

```
Aligning 03bSnISJMiM[110]:   0%|          | 0/1 [00:00<?, ? Segments/s]Traceback (most recent call last):
  File "/Users/krayush/cmu_multimodal/CMU-MultimodalSDK/temp_align.py", line 14, in <module>
    cmumosi_highlevel.align('Opinion Segment Labels')
  File "/Users/krayush/cmu_multimodal/CMU-MultimodalSDK/mmsdk/mmdatasdk/dataset/dataset.py", line 121, in align
    intersects,intersects_features=self.__intersect_and_copy(ref_time,relevant_entries[otherseq_key][entry_key],epsilon)
KeyError: '03bSnISJMiM[110]'
```

The code being run:

```
cmumosi_highlevel.align('glove_vectors', collapse_functions=[myavg])
cmumosi_highlevel.align('Opinion Segment Labels')

deploy_files = {x: x for x in cmumosi_highlevel.computational_sequences.keys()}
cmumosi_highlevel.deploy("./deployed", deploy_files)
```

PS: I get no error when I run `cmumosi_highlevel.align('Opinion Segment Labels')` and `cmumosi_highlevel.align('glove_vectors', collapse_functions=[myavg])` independently.

**Comment by A2Zadeh** [comment]
I am looking into this now. Will give updates, could be a bug on the newest words.

**Comment by A2Zadeh** [comment]
Ok so if you look at your computational sequences one of them does not have: 03bSnISJMiM[110]

This can happen if the detector fails (like bad mic, or occluded face). It's up to you to replace it with some imputated values, i.e. zeros. The intervals should be the same as glove 03bSnISJMiM[110] intervals, the features should be the same shape as the detectors (for example 1x72 for COVAREP) but all zeros. Does this answer the question? 

We don't normally imputate the data since it is a choice that should be made by the machine learning scientist.

**Comment by krayush07** [comment]
I still couldn't get why do I not get the error when running them independently and is there any sample code already to do the process which you mentioned?

**Comment by krayush07** [comment]
@A2Zadeh This error is not because of the reason you mentioned, rather some issue with the alignment that I am doing.

My entry key is `03bSnISJMiM[110]` but `relevant_entries[otherseq_key]` contains keys equivalent to video ids i.e, `03bSnISJMiM` and so on.

**Comment by A2Zadeh** [comment]
I see. Can you help me out to recreate your problem on my side? What are the computational sequences you are using? I am not getting similar issue on my side. Thanks for your patience.

**Comment by krayush07** [comment]
These are the computational sequences I am using:

```
CMU_MOSI_OpenSmile_EB10.csd
CMU_MOSI_Opinion_Labels.csd
CMU_MOSI_TimestampedBERT.csd
CMU_MOSI_TimestampedWordVectors.csd
```

And the alignment sequence is:

```
# not sure to collapse or not for 1st alignment
cmumosi_highlevel.align('glove_vectors')
cmumosi_highlevel.align('Opinion Segment Labels')


deploy_files = {x: x for x in cmumosi_highlevel.computational_sequences.keys()}
cmumosi_highlevel.deploy("./deployed", deploy_files)
```

**Comment by A2Zadeh** [comment]
@krayush07 the code you sent has a major flaw. You should not include the Opinion Segments in alignment of words. In simple terms you should only add opinion segments after alignment:
start with

CMU_MOSI_OpenSmile_EB10.csd
CMU_MOSI_TimestampedBERT.csd
CMU_MOSI_TimestampedWordVectors.csd

Then align using 
cmumosi_highlevel.align('glove_vectors') also use collapse.

Then add the computational sequence CMU_MOSI_Opinion_Labels.csd to the dataset using:

cmumosi_highlevel.add_computational_sequences(mmdatasdk.cmu_mosi.labels,'cmumosi/')

and then align to opinion labels. 

In simple terms follow this example to the book: 

https://github.com/A2Zadeh/CMU-MultimodalSDK/blob/master/examples/mmdatasdk_examples/word_level_align.py
